lib/spack/docs/repositories.rst

```diff
@@ -0,0 +1,275 @@
+.. _repositories:
+
+=============================
+Local Package Repositories
+=============================
+
+Spack comes with over 1,000 builtin package recipes in the
+``var/spack/repos/builtin/`` directory.  This is a **package
+repository**, a directory that Spack searches when it needs to find a
+package by name.  By default, the :ref:`cmd-spack-edit` and
+:ref:`cmd-spack-create` commands modify packages in the builtin
+repository.  You may need to maintain a set of packages **separate** from
+the builtin repository, for private, proprietary, export-controlled, or
+otherwise restricted software. Spack allows you to configure local
+repositories using either the ``repos.yaml`` or the ``spack repo``
+command.
+
+A package repository is simply a directory structured like this::
+
+  repo/
+      repo.yaml
+      packages/
+          hdf5/
+              package.py
+          mpich/
+              package.py
+              mpich-1.9-bugfix.patch
+          trilinos/
+              package.py
+          ...
+
+The top-level ``repo.yaml`` file contains configuration metadata for the
+repository, and the ``packages`` directory contains subdirectories for
+each package in the repository.  Each package directory contains a
+``package.py`` file and any patches or other files needed to build the
+package.
+
+
+---------------------
+``repos.yaml``
+---------------------
+
+Spack finds repositories by looking at the ``repos.yaml`` configuration
+file.  For more on the YAML format, and on how configuration file
```

Comment:
Looking at or looking for?  Oh, I see my confusion... `repos.yaml` is in the configuration area, but `repo.yaml` is at the top of every repo.  This is a "gotcha" that could probably be clarified by a carefully placed sentence in the docs.